docs: add initial materialize-hubspot docs (#3042)
estuary/flow: site/docs/reference/Connectors/materialization-connectors/hubspot.md

---

> description: Materialize collections to HubSpot CRM objects such as Contacts, Companies, Deals, etc.

Reviewer: ```suggestion
description: Materialize collections to HubSpot CRM objects such as Contacts, Companies, and Deals for reverse-ETL workflows.
```
Nitpicky and may just be personal preference: to me, "etc" feels like we're trailing off. The "such as" should already imply there are more objects to the list.

---

# HubSpot

The HubSpot connector writes to HubSpot CRM objects such as Contacts,

> Companies, Deals, etc.

Reviewer: ```suggestion
Companies, and Deals.
```

connector. To match incoming documents to existing records, the connector
compares the HubSpot record against the **collection key**.

:::important
Using a **unique** property as the match key is strongly recommended.

Non-unique properties require a search to determine whether a record exists.
Due to HubSpot search API limitations, using a non-unique match property can
result in duplicate records.
:::

To determine if a property is unique, navigate in the HubSpot webapp to `Settings
/ Data Management / Properties` and check in the Property Rules if "require unique
values for this property" is enabled.  Alternatively, look at the [object
definition][] or use the [Properties API][] to list items with `hasUniqueValue`
set to `true`.

If there is no existing unique property that is suitable, it is recommended to
create a new property with "require unique values" set.

## Property name mapping

Collection field names are mapped to HubSpot property names automatically
similarly to how HubSpot generates property names: the field name is
lowercased, leading underscores are removed as well as other symbols
characters, and if the field begins with a number it will be prefixed with

> `n`.Names are truncated to 100 characters.

Reviewer: ```suggestion
`n`. Names are truncated to 100 characters.
```
Missing space?
